superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1509,6 +1511,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
       show={show}
       title={<h4>{t('Connect a database')}</h4>}
       footer={renderModalFooter()}
+      hideFooter={renderModalFooter() ? false : true}
```

Comment:
You can use simpler truthy/falsey statement here with the existing `modalFooter` variable:
`hideFooter={!modalFooter}`